### ISSUE TITLE
Need to connect to web socket server with cookie

### ISSUE THREAD

**Issue by djarvis** [chkr1011/MQTTnet#495]
I need to connect to a websocket server with the outgoing initial HTTP request containing a cookie.  Is this possible?  I didn't see any options in MqttClientOptionsBuilder.

**Comment by chkr1011** [comment]
I believe this is related to #496.

**Comment by djarvis** [comment]
I was able to do this by creating a MqttClientWebSocketOptions class and attaching a CookieContainer, then assigning this to ChannelOptions of a manually created MqttClientOptions class.

In other words this is possible without using the builder, but it'd be nice if there was an override that included a cookie container on WithWebSocketServer() extension.

**Comment by cquesnel** [comment]
Not sure, if cookies are passed through request headers, but while adding the option for headers, I will also add an overload for the `CookieContainer`. As @djarvis says, a `MqttClientWebSocketOptions` class could be passed in the builder the same way as is done with `MqttClientOptionsBuilderTlsParameters`.

All while leaving the default call with only the url for backwards compatibility

**Comment by djarvis** [comment]
I just tested this and I do get cookies back to the client from the connect call, and when I set them they are also recognized on the server.

I am doing this because, in scaled Azure instances, I wish to target specific instances of my MQTT broker behind the load balancer.  I can tell which instance the client is talking to by observing the ARRAffinity cookie in my cookie collection, and I can also target a specific instance to connect with by setting the cookie.
